### ISSUE TITLE
Better plots for TTim

### ISSUE THREAD

**Issue by dbrakenhoff** [mbakker7/ttim#70]
The plot functions under the model object in TTim are somewhat limited. 

Proposal:
- move to a plotting submodule (like pastas). `ml.plots.topview()` `ml.plots.xsection()` instead of having all plot functions directly in the model class.
- keep `ml.plot()`: plot top-view of model with optional layer arg
- modify `ml.xsection()` to plot cross-section of model 
- the current `ml.xsection()` becomes something like `ml.plot.heads_along_line()`

Other suggestions welcome!

**Comment by dbrakenhoff** [comment]
Example for a cross-section:
```python
ml.plots.xsection(labels=True, params=True)
```
![image](https://github.com/user-attachments/assets/1e6c329b-d980-44fe-a800-fad455bfa554)

**Comment by dbrakenhoff** [comment]
Good suggestions.
- returning `ax` is definitely something we want (figure is always available through `ax.figure`)

> - plot location and property of elements as well (e.g. LineSinks)

This is a bit of a challenge, though definitely doable. We'd need to compute intersections for all elements the line crosses. And give each element a `e.plot_xsection()` method that takes some distance `r` as input and plots the vertical representation of the element at that location. Giving each element a shapely representation through `e.geometry` would make that fairly easy though (as suggested here: https://github.com/mbakker7/timml/issues/69). 

> Seperate plotly part, or plot function that returns x and y lists of data to plot

What would you need for these plots? For now you could add a tim_plotly repository to the [tim groundwater organization](https://github.com/tim-groundwater). With methods that take a Tim-model as input and produce some kind of plotly plot.  

> - highlight aquifer that is plotted, when plotting in a section,

What do you mean by this one?

**Comment by mbakker7** [comment]
Nice ideas. I suggest to start with a plots module and add some functionality there.

**Comment by HMEUW** [comment]
> highlight aquifer that is plotted, when plotting in a section: What do you mean by this one?

See example below. Lines are calculated heads in aquifer 1 and 2. The aquifers are filled to indicate in which aquifer the calculated heads are present. Or dashed line in the center of the aquifer.

![image](https://github.com/user-attachments/assets/131748da-0445-47ec-b3eb-9e9c97821c99)

![image](https://github.com/user-attachments/assets/95ca2bc2-b43c-4234-92a7-710130bd9785)

**Comment by dbrakenhoff** [comment]
This was implemented in #76. Closing this issue. 

The aquifer cross-section and heads plot can be easily done with:

```python
xy = [(x0, y0), (x1, y1)]
ax = ml.plots.xsection(xy)
ml.plots.headalongline(x0, x1, y0, y1, ax=ax)
ax.legend(loc=(0, 1), frameon=False, ncol=4)
```
